torch/fx/experimental/proxy_tensor.py

```diff
@@ -679,6 +684,17 @@ def set_original_aten_op(func):
         yield
 
 
+class TorchFunctionMetadataMode(TorchFunctionMode):
```

Comment:
I wonder if there is any value of getting this metadata from dynamo directly in strict mode? Or maybe this is better because it is consistent between strict and not-strict

Reply:
Something similar that dynamo produces is basically what we see in `source_fn_stack` [here](https://fburl.com/code/5iwhiyr3). Except it's kind of different in that for for module calls it'll point to the actual module, like `torch.nn.Linear` rather than the torch.function, `torch.nn.functional.linear`. I feel like this one (`torch.nn.functional.linear`) is better.